Quickstart k8s otel doc added (#4488)
elastic/observability-docs: docs/en/observability/quickstarts/monitor-k8s-otel.asciidoc

[[monitor-k8s-otel-edot]]
= Quickstart: Unified Kubernetes Observability with Elastic Distributions of OpenTelemetry (EDOT)

preview::[]

> In this quickstart guide, you will learn how to send Kubernetes logs, metrics, and application traces to Elasticsearch using the https://github.com/open-telemetry/opentelemetry-operator/[OpenTelemetry Operator] to orchestrate the https://github.com/elastic/opentelemetry/tree/main[Elastic Distributions of OpenTelemetry] (EDOT) Collectors and SDK instances you will need.

Reviewer: This sentence is a little confusing. I think it's the `you will need` at the end that's throwing me off. I wonder if a list might work better here? Something like:

In this quickstart guide, you will learn:
* how to send Kubernetes logs, metrics, and application traces to Elasticsearch with the [OpenTelemetry Operator](https://github.com/open-telemetry/opentelemetry-operator/)
* how to orchestrate the [Elastic Distributions of OpenTelemetry](https://github.com/elastic/opentelemetry/tree/main) (EDOT) Collectors and SDK instances with the [OpenTelemetry Operator](https://github.com/open-telemetry/opentelemetry-operator/).

Author: I have simplified it a bit with:

> In this quickstart guide, you will learn how to send Kubernetes logs, metrics, and application traces to Elasticsearch using the https://github.com/open-telemetry/opentelemetry-operator/[OpenTelemetry Operator] to orchestrate https://github.com/elastic/opentelemetry/tree/main[Elastic Distributions of OpenTelemetry] (EDOT) Collectors and SDK instances.

The reasons I prefer to not add a list of two items are:
- both items are actually the same.
- about second idem the user won't really learn specifically `how to orchestrate collectors and SDKs with the operator`. They will learn how to deploy all components at once basically (with all pre-configured).

Let me know your thoughts ;)

Reviewer: Perfect. Thanks for the explanation!

All the components will be deployed through the https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-kube-stack[opentelemetry-kube-stack] helm chart. They include:

* https://github.com/open-telemetry/opentelemetry-operator/[OpenTelemetry Operator]
* `DaemonSet` EDOT Collector configured for node level metrics.
* `Deployment` EDOT Collector configured for cluster level metrics.

> * Instrumentation object for applications https://opentelemetry.io/docs/kubernetes/operator/automatic/[auto-instrumentation].

Reviewer: There is inconsistent punctuation used in this list. Maybe add a period to the first item

For a more detailed description of the components and advanced configuration refer to the https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/README.md[elastic/opentelemetry] GitHub repository.

[discrete]
== Prerequisites

> * An {es} cluster, self-managed or a hosted {ess} deployment on {ess-trial}[{ecloud}]. The deployment includes an {es} cluster for storing and searching your data, and {kib} for visualizing and managing your data.

Reviewer: This text is slightly different from the text we use in the other quick start guides. Let's match the text we use in those guides for now. If you think it should be updated, we should do it in all three. We can open an issue to address that.

Author: I was assuming `self-managed` clusters could also use the onboarding UX, but honestly I haven't checked.
I'm ok about leaving the same text as in the other guides.

* A running Kubernetes cluster (v1.23 or newer).
* https://kubernetes.io/docs/reference/kubectl/[Kubectl].
* https://helm.sh/docs/intro/install/[Helm].
* (optional) https://cert-manager.io/docs/installation/[Cert-manager], if you opt for automatic generation and renewal of publicly trusted TLS certificates.

[discrete]
== Collect your data

.  In {kib}, go to **Observability** and click **Add Data**.

. Under the question *`What do you want to monitor?`* select **Kubernetes**, and then select the **OpenTelemetry: Full Observability** option.
+
[role="screenshot"]
image::images/quickstart-k8s-otel-entry-point.png[Kubernetes-OTel entry point]

. Follow the on-screen instructions to perform the installation of all needed components.
+
[NOTE]
====

> The default installation deploys the OpenTelemetry Operator with a self-signed TLS certificate. For automatic certificate renewal with https://cert-manager.io/docs/installation/[cert-manager], refer to https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/README.md#cert-manager[cert-manager integrated installation] for instructions on customizing the `values.yaml` file before running the `helm install` command.

Reviewer: This sentence also feels a little long. Maybe we can break it up. Something like:

The default installation deploys the OpenTelemetry Operator with a self-signed TLS certificate. Refer to the cert-manager integrated installation guide to enable automatic certificate renewal using cert-manager. This guide provides instructions on customizing the values.yaml file before running the helm install command.

Author: Thanks! It looks much better... I have actually removed the final part `This guide provides instructions on customizing the values.yaml file before running the helm install command.`, as it's probably unneeded. Let me know your thoughts. It will look like:

> The default installation deploys the OpenTelemetry Operator with a self-signed TLS certificate. Refer to the https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/README.md#cert-manager[cert-manager integrated installation] guide to enable automatic certificate renewal using https://cert-manager.io/docs/installation/[cert-manager].

Author: @bmorelli25 , I had to update a bit the info about the TLS certificate in a last minute commit, as cert-manager in this case is used for automatic generation (AND RENEWAL) of the certificate (still self-signed but automatically renewed).
The information about publicly trusted certs was incorrect, although that's a very common use case for `cert-manager`.

would you please review?

Reviewer: This edit looks good too!

====
+
Install the OpenTelemetry Operator using the kube-stack Helm chart and the provided values file. You will run a few commands to:
+ 
 * Add the helm chart repository needed for the installation.
 * Create a namespace.
 * Create a secret with an API Key and the {es} endpoint to be used by the collectors.
 * Deploy the `opentelemetry-kube-stack` helm chart with the provided `values.yaml`.

[discrete]
== Visualize your data

After installation is complete and all relevant data is flowing into Elastic,

> the **Visualize your data** section provides a link to the dashboard used to monitor the health of the cluster (*[OTEL][Metrics Kubernetes]Cluster Overview*).

Reviewer: I would move this out of parenthesis by using the dashboard name in the sentence:

After installation is complete and all relevant data is flowing into Elastic,
the **Visualize your data** section provides a link to the *[OTEL][Metrics Kubernetes]Cluster Overview* dashboard used to monitor the health of the cluster.

Author: The reason of adding it in the parenthesis at the end of the sentence was to reduce the importance of the exact name of the dashboard, as I'm expecting this to change soon or extra dashboards and assets added.

But I'm ok with the change, it looks better!

[role="screenshot"]
image::images/quickstart-k8s-otel-dashboard.png[Kubernetes overview dashboard]

[discrete]
== Troubleshooting and more

> * For troubleshooting the components deployment and installation refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#installation-verification[installation verification].

Reviewer: Simplify to this?

```suggestion
* For troubleshooting deployment and installation, refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#installation-verification[installation verification].
```

> * For applications instrumentation refer to https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/instrumenting-applications.md[Instrumenting applications with EDOT SDKs on Kubernetes].
> * For customizing the configuration refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#custom-configuration[custom configuration] (TBD).
> * Refer to <<observability-introduction>> for a description of other useful features.

Reviewer: ```suggestion
* For applications instrumentation, refer to https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/instrumenting-applications.md[Instrumenting applications with EDOT SDKs on Kubernetes].
* For customizing the configuration, refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#custom-configuration[custom configuration] (TBD).
* Refer to <<observability-introduction>> for a description of other useful features.
```

[[monitor-k8s-otel-edot]]
= Quickstart: Unified Kubernetes Observability with Elastic Distributions of OpenTelemetry (EDOT)

preview::[]

In this quickstart guide, you will learn how to send Kubernetes logs, metrics, and application traces to Elasticsearch, using the https://github.com/open-telemetry/opentelemetry-operator/[OpenTelemetry Operator] to orchestrate https://github.com/elastic/opentelemetry/tree/main[Elastic Distributions of OpenTelemetry] (EDOT) Collectors and SDK instances.

All the components will be deployed through the https://github.com/open-telemetry/opentelemetry-helm-charts/tree/main/charts/opentelemetry-kube-stack[opentelemetry-kube-stack] helm chart. They include:

* https://github.com/open-telemetry/opentelemetry-operator/[OpenTelemetry Operator].
* `DaemonSet` EDOT Collector configured for node level metrics.
* `Deployment` EDOT Collector configured for cluster level metrics.
* `Instrumentation` object for applications https://opentelemetry.io/docs/kubernetes/operator/automatic/[auto-instrumentation].

> For a more detailed description of the components and advanced configuration refer to the https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/README.md[elastic/opentelemetry] GitHub repository.

Reviewer 2: ```suggestion
For a more detailed description of the components and advanced configuration, refer to the https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/README.md[elastic/opentelemetry] GitHub repository.
```

[discrete]
== Prerequisites

* A deployment using our hosted {ess} on {ess-trial}[{ecloud}]. The deployment includes an {es} cluster for storing and searching your data, and {kib} for visualizing and managing your data.
* A running Kubernetes cluster (v1.23 or newer).
* https://kubernetes.io/docs/reference/kubectl/[Kubectl].
* https://helm.sh/docs/intro/install/[Helm].
* (optional) https://cert-manager.io/docs/installation/[Cert-manager], if you opt for automatic generation and renewal of TLS certificates.

[discrete]
== Collect your data

.  In {kib}, go to **Observability** and click **Add Data**.

> . Under the question *`What do you want to monitor?`* select **Kubernetes**, and then select the **OpenTelemetry: Full Observability** option.

Reviewer 2: ```suggestion
. Under *`What do you want to monitor?`*, select **Kubernetes**, and then select the **OpenTelemetry: Full Observability** option.
```

+
[role="screenshot"]
image::images/quickstart-k8s-otel-entry-point.png[Kubernetes-OTel entry point]

> . Follow the on-screen instructions to perform the installation of all needed components.

Reviewer 2: ```suggestion
. Follow the on-screen instructions to install all needed components.
```

+
[NOTE]
====
The default installation deploys the OpenTelemetry Operator with a self-signed TLS certificate valid for 365 days. This certificate **won't be renewed** unless the Helm Chart release is manually updated. Refer to the https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/README.md#cert-manager[cert-manager integrated installation] guide to enable automatic certificate generation and renewal using https://cert-manager.io/docs/installation/[cert-manager].
====
+
Deploy the OpenTelemetry Operator and EDOT Collectors using the kube-stack Helm chart with the provided `values.yaml` file. You will run a few commands to:
+ 
 * Add the helm chart repository needed for the installation.
 * Create a namespace.
 * Create a secret with an API Key and the {es} endpoint to be used by the collectors.
 * Install the `opentelemetry-kube-stack` helm chart with the provided `values.yaml`.

>  * Optionally, for instrumenting applications, apply the corresponding `annotations` as showed in {kib}.

Reviewer 2: ```suggestion
 * Optionally, for instrumenting applications, apply the corresponding `annotations` as shown in {kib}.
```

[discrete]
== Visualize your data

After installation is complete and all relevant data is flowing into Elastic,
the **Visualize your data** section provides a link to the *[OTEL][Metrics Kubernetes]Cluster Overview* dashboard used to monitor the health of the cluster.

[role="screenshot"]
image::images/quickstart-k8s-otel-dashboard.png[Kubernetes overview dashboard]

[discrete]
== Troubleshooting and more

* For troubleshooting deployment and installation, refer to https://github.com/elastic/opentelemetry/tree/main/docs/kubernetes/operator#installation-verification[installation verification].

> * For applications instrumentation details, refer to https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/instrumenting-applications.md[Instrumenting applications with EDOT SDKs on Kubernetes].

Reviewer 2: ```suggestion
* For application instrumentation details, refer to https://github.com/elastic/opentelemetry/blob/main/docs/kubernetes/operator/instrumenting-applications.md[Instrumenting applications with EDOT SDKs on Kubernetes].
```
